File: Examples/Features/Workbook/ImportExcelFileDataOnly.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace GrapeCity.Documents.Excel.Examples.Features.Workbook
{
    public class ImportExcelFileDataOnly : ExampleBase
    {
        public override void Execute(GrapeCity.Documents.Excel.Workbook workbook)
        {
            //Use XlsxOpenOptions.ImportFlags to control what you want to import from excel, ImportFlags.Data means only the data will be imported
            //Change the path to the real file path when open.
            XlsxOpenOptions options = new XlsxOpenOptions();
            options.ImportFlags = ImportFlags.Data;

            workbook.Open(this.CurrentDirectory + "source.xlsx", options);

        }

        public override bool CanDownload
        {
            get
            {
                return false;
            }
        }

        public override bool ShowViewer
        {
            get
            {
                return false;
            }
        }

        public override bool IsUpdate
        {
            get
            {
                return true;
            }
        }
    }
}

File: Examples.VB/Features/Workbook/ImportExcelFileDataOnly.vb
Namespace Features.Workbook
    Public Class ImportExcelFileDataOnly
        Inherits ExampleBase
        Public Overrides Sub Execute(workbook As Excel.Workbook)
            'Use XlsxOpenOptions.ImportFlags to control what you want to import from excel, ImportFlags.Data means only the data will be imported
            'Change the path to the real file path when open.
            Dim options As New XlsxOpenOptions With {
                .ImportFlags = ImportFlags.Data
            }
            workbook.Open(IO.Path.Combine(CurrentDirectory, "source.xlsx"), options)
        End Sub
        Public Overrides ReadOnly Property CanDownload As Boolean
            Get
                Return False
            End Get
        End Property
        Public Overrides ReadOnly Property ShowViewer As Boolean
            Get
                Return False
            End Get
        End Property
        Public Overrides ReadOnly Property IsUpdate As Boolean
            Get
                Return True
            End Get
        End Property
    End Class
End Namespace
